libsession/src/main/java/org/session/libsession/messaging/jobs/BatchMessageReceiveJob.kt

```diff
@@ -150,14 +165,15 @@ class BatchMessageReceiveJob(
                             }
                         }
                         // increment unreads, notify, and update thread
-                        val unreadFromMine = messageIds.indexOfLast { (_,fromMe) -> fromMe }
-                        var trueUnreadCount = messageIds.filter { (_,fromMe) -> !fromMe }.size
+                        val unreadFromMine = messageIds.map { it.value.first }.indexOfLast { it }
+                        var trueUnreadCount = messageIds.filter { !it.value.first }.size
+                        val trueUnreadMentionCount = messageIds.filter { !it.value.first && it.value.second }.size
                         if (unreadFromMine >= 0) {
                             trueUnreadCount -= (unreadFromMine + 1)
```

Comment:
I'm guessing this might not have any new issues relating to the unread mention count? For example showing an unread mention in the same batch but having a message from our own device afterwards? But if it looks good on your end probably nothing to worry about

Reply:
Yea I'm not a fan of caching the unread count as a value in the database but I don't think we'd be able to query it live without a negative performance impact

I'll push up logic to update the `trueUnreadMentionCount` if we go within the `unreadFromMine` statement